### ISSUE TITLE
Compile error : simple_key_value_store - error: variable does not need to be mutable  (rust nightly)

### ISSUE THREAD

**Issue by rossmuir** [maidsafe/routing#558]
Ubuntu 15.04 / Rust version:
```
rustc 1.3.0-nightly (38517944f 2015-08-03)
cargo 0.4.0-nightly (1b902d4 2015-08-03) (built 2015-08-03)
```
Compilation error
```
Compiling routing v0.2.8 (file:///home/user/MaidSafe-Testing/maidsafe_rust/nightly-project/routing)
examples/simple_key_value_store.rs:374:9: 374:18 error: variable does not need to be mutable
examples/simple_key_value_store.rs:374     let mut stdin = io::stdin();
                                               ^~~~~~~~~
examples/simple_key_value_store.rs:26:62: 26:68 note: lint level defined here
examples/simple_key_value_store.rs:26         unconditional_recursion, unknown_lints, unsafe_code, unused,
                                                                                                   ^~~~~~
error: aborting due to previous error
Build failed, waiting for other jobs to finish...
Could not compile `routing`.
```

**Comment by benjaminbollen** [comment]
I'm on an old nightly `rustc 1.2.0-nightly (31d9aee68 2015-06-22)`, but I don't get this error on master, and if I make the variable immutable, then I get the error that it needs to be mutable.

I'll have to update my nightly first to get in sync with this error... (as a weird but my best bet)

**Comment by benjaminbollen** [comment]
fix in PR #559

**Comment by rossmuir** [comment]
Thank you
